odoo/addons/base/models/ir_attachment.py

```diff
@@ -815,3 +815,17 @@ def _to_http_stream(self):
             stream.size = 0
 
         return stream
+
+    def action_x509_extract_public_key(self):
+        ctx = dict(self.env.context)
+        res_id = ctx.pop('res_id')
+        res_model = ctx.pop('res_model')
+        self = self.with_context(ctx)
```

Comment:
Merge with the following line, it is a bad practice to redefine `self`. (ruff will also complain)